packages/apollo-link-state/src/index.ts

```diff
@@ -57,7 +57,7 @@ export const withClientState = (
         resolvers[(rootValue as any).__typename || type][
           info.resultKey || fieldName
         ];
-      if (resolve) return resolve(rootValue, args, context, info);
+      if (resolve) return resolve(rootValue, args, context, info) || null;
```

Comment:
I would add a check for only `undefined`, because the resolver might return `0` or `false` by design.

Reply:
And what if it's an async resolver? A promise will always be truthy, even if it resolves to undefined.